Improve styles for sensei login page

Imran92 opened the pull request:
Resolves https://github.com/Automattic/sensei/issues/6368

## Proposed Changes
* Improved the style of the login page following the Woocommerce login
* Improved the alert message shown when there's a login error
* Fixed the issue of "My Messages" button getting rendered even for logged-out users

## Testing Instructions
<!--
Add as many details as possible to help others reproduce the issue and test the changes.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1. Log out
2. Go to My Courses page from frontend
3. It should look better than before and nothing should be broken
4. Try with all variations of the Course theme, Divi and Astra
5. Enable registration from Settings -> General -> Anyone can Register
6. Make sure it looks better with the Registration form too
7. Check in mobile view as well for all the above
8. Check the error alert with a wrong password

#### Course theme:
<img width="1113" alt="Screenshot 2023-10-05 at 4 32 17 PM" src="https://github.com/Automattic/sensei/assets/6820724/6f39930b-8265-418b-82c1-873d7e4f336e">
<img width="1113" alt="Screenshot 2023-10-05 at 4 32 37 PM" src="https://github.com/Automattic/sensei/assets/6820724/88c6eaf3-79ab-4fde-a779-49ace8df16e2">
<img width="1113" alt="Screenshot 2023-10-05 at 4 32 52 PM" src="https://github.com/Automattic/sensei/assets/6820724/4e722634-f628-422d-98f9-b23b5073ed68">
<img width="1113" alt="Screenshot 2023-10-05 at 4 33 09 PM" src="https://github.com/Automattic/sensei/assets/6820724/ac70dfcc-90fa-4601-9c49-ba0b64c7b823">

#### Divi
<img width="1191" alt="Screenshot 2023-10-05 at 4 34 43 PM" src="https://github.com/Automattic/sensei/assets/6820724/dc9bc27f-993e-49ed-8af3-913f1f749c02">

#### Astra
<img width="1249" alt="Screenshot 2023-10-05 at 4 34 07 PM" src="https://github.com/Automattic/sensei/assets/6820724/a5a4fb6a-ef42-429d-a508-84c10958af21">

#### Mobile
<img width="523" alt="Screenshot 2023-10-05 at 4 35 34 PM" src="https://github.com/Automattic/sensei/assets/6820724/cc1b83ad-b6a3-4be1-98a0-160b3b47ed0d">

#### Error message
![image (2)](https://github.com/Automattic/sensei/assets/6820724/0029010f-b3b0-4fea-a504-ce4774935912)

## Pre-Merge Checklist
<!-- Complete applicable items on this checklist **before** merging. Items that are not applicable can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed. -->
- [x] PR title and description contain sufficient detail and accurately describe the changes
- [x] Acceptance criteria is met
- [x] Decisions are publicly documented
- [x] Adheres to coding standards ([PHP](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/), [JavaScript](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/javascript/), [CSS](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/css/), [HTML](https://developer.wordpress.org/coding-standards/wordpress-coding-standards/html/))
- [ ] All strings are translatable (without concatenation, handles plurals)
- [x] Follows our naming conventions (P6rkRX-4oA-p2)
- [ ] Hooks (p6rkRX-1uS-p2) and functions are documented
- [x] New UIs are responsive and use a [mobile-first approach](https://zellwk.com/blog/how-to-write-mobile-first-css/)
- [x] New UIs match the designs
- [x] Different user privileges (admin, teacher, subscriber) are tested as appropriate
- [x] Code is tested on the minimum supported PHP and WordPress versions
- [x] User interface changes have been tested on the latest versions of Chrome, Firefox and Safari
- [ ] "Needs Documentation" label is added if this change requires updates to documentation
- [ ] Known issues are created as new GitHub issues

Imran92 committed:
Fix my messages block rendering when not logged in

Imran92 committed:
add additional existing classes to fix input field and button

Imran92 committed:
Fix register button

Imran92 committed:
Fix login and registration form styles

Imran92 committed:
Add changelog

Imran92 commented on templates/user/login-form.php:
```diff
@@ -39,15 +39,15 @@
 				<label for="sensei_user_login"><?php esc_html_e( 'Username or Email', 'sensei-lms' ); ?> </label>
 
-				<input type="text" name="log" id="sensei_user_login" class="input" value="" size="20">
+				<input type="text" name="log" id="sensei_user_login" class="input input-text" value="" size="20">
```
I didn't name `input-text` as per the convention because the same class name is used in the registration form. It's also looks like a convention to use this class in form input text fields, also used in woocommerce forms.

Imran92 committed:
Update template version

Imran92 committed:
Add development changelog

Imran92 committed:
Fix alert message style

renatho commented:
It's getting waaay better! 🎉

But I noticed a few things:

### Different pages

- When I set up a new site, the default My Course page is using the "Course List" variation, and the new styles aren't being applied there.
- We also have the legacy My Courses with the `[sensei_user_courses]` shortcode where the new styles aren't being applied. Do we want to also improve the login there?

Maybe we could just use another class instead of the `wp-block-sensei-lms-learner-courses` to make it work for all the cases? If yes, I'd suggest adding a new class where we have the `my-courses` ID in `sensei_login_form`, just to not use the ID to encapsulate styles.

For discussion: Not part of this issue, but should we deprecate the Student Courses block? Or are we still using or encouraging it somehow?

### Other styles

Changing the class manually to test the `[sensei_user_courses]` shortcode, I noticed it has a `width: 90%` on the inputs coming from `assets/css/frontend.scss` (see following screenshot), which also adds some styles to the login/registration form 🤔. Should we just tweak this? Or maybe try to remove those styles completely? I didn't investigate to see what is the reason for the 2 different files. Any idea?

<img width="702" alt="Screenshot 2023-10-10 at 14 39 42" src="https://github.com/Automattic/sensei/assets/876340/8b30f87e-e181-4335-af72-fd03991a1159">

### Error in the wrong place in some themes

With Divi and Twenty Twenty-One, I noticed that the error is displayed in the wrong place. See following screenshots.

<img width="1725" alt="Screenshot 2023-10-10 at 14 46 41" src="https://github.com/Automattic/sensei/assets/876340/e224da60-b512-4d45-96a6-572a8b142fd4">

In Divi, I could see it after scrolling, otherwise the error is displayed behind the header

<img width="1198" alt="Screenshot 2023-10-10 at 15 15 18" src="https://github.com/Automattic/sensei/assets/876340/0ff47181-c0a3-4275-9522-c21ae06324ea">

### Smaller screens

In Twenty Twenty-One, Two and Three, I noticed that it doesn't have enough space, so the "Lost your password?" is breaking the line in a weird way. Maybe we could improve in a way that it breaks the line in the link space only?

<img width="411" alt="Screenshot 2023-10-10 at 14 43 47" src="https://github.com/Automattic/sensei/assets/876340/d946c6a6-ee81-4c7c-b9f5-f34b6c9adeb4">

Imran92 committed:
Fix form getting shrunk in twenty twenty themes

Imran92 committed:
Use existing div id as style selector

This covers also the non-block scenarios like shortcode and hook based renderings of login form, like with the course list block

Imran92 committed:
Remove unused styles

Imran92 committed:
Fix alert going behind header in divi

Imran92 committed:
Show login errors above login form

Imran92 commented:
Hi @renatho , thanks so much for reviewing -

> When I set up a new site, the default My Course page is using the "Course List" variation, and the new styles aren't being applied there.
> We also have the legacy My Courses with the [sensei_user_courses] shortcode where the new styles aren't being applied. Do we want to also improve the login there?
> Maybe we could just use another class instead of the wp-block-sensei-lms-learner-courses to make it work for all the cases? If yes, I'd suggest adding a new class where we have the my-courses ID in sensei_login_form, just to not use the ID to encapsulate styles.

I've just fallen back to use the previous my-courses ID selector as it is already there. It has solved issues with both the login shortcode and course list case https://github.com/Automattic/sensei/pull/7204/commits/e260a0dd1934c6e6821bfad902c1cb2b9876be7c .

>  I noticed it has a width: 90% on the inputs coming from assets/css/frontend.scss

Yap right, thanks. It was some now-stale CSS. I've removed it here https://github.com/Automattic/sensei/pull/7204/commits/22bdc7abe8304ac1d0f5176bab31c0d1b1a0b54e

> In Twenty Twenty-One, Two and Three, I noticed that it doesn't have enough space, so the "Lost your password?" is breaking the line in a weird way. Maybe we could improve in a way that it breaks the line in the link space only?

Actually, in all the TwentyTwenty series themes, the whole content was getting shrunk for some styles in those themes making the whole page look bad. So I've fixed it here https://github.com/Automattic/sensei/pull/7204/commits/e260a0dd1934c6e6821bfad902c1cb2b9876be7c and now it has the same width as the woocommerce login page.

> With Divi and Twenty Twenty-One, I noticed that the error is displayed in the wrong place. See following screenshots.

I did a bit of investigation and it turns out that the issue was not only with Divi and Twenty Twenty One, rather it was in all non-block themes. The notices got printed on top of the page, even above the navbar or header. It was due to how we handle the notices in the template. I've added a fix https://github.com/Automattic/sensei/pull/7204/commits/7f7270de5d793fa65dfb7a56b63575ddacad91c5 which makes the notices load nicely in all themes now.

Divi:
![image](https://github.com/Automattic/sensei/assets/6820724/6136b1d1-cc28-4ea9-bf40-438e2e914be0)

Twenty Twenty One:
![image](https://github.com/Automattic/sensei/assets/6820724/b9452267-76d5-4263-958d-ecf9631fbb25)

Thanks!

Imran92 committed:
Merge branch 'trunk' into add/styles-for-sensei-login-page

renatho commented:
Thank you for the fixes, @Imran92!

> I did a bit of investigation and it turns out that the issue was not only with Divi and Twenty Twenty One, rather it was in all non-block themes. The notices got printed on top of the page, even above the navbar or header. It was due to how we handle the notices in the template. I've added a fix https://github.com/Automattic/sensei/commit/7f7270de5d793fa65dfb7a56b63575ddacad91c5 which makes the notices load nicely in all themes now.

It seems it only fixes it for this specific notice, but the others using this method continue with the issue. I found https://github.com/Automattic/sensei/issues/6173. If it's not very easy to fix, maybe we could revert the commit and work on that in another PR?

---

About Twenty Twenty-One, I still noticed that we don't have spacing on the buttons. But feel free to ignore this one if it's not easy or needs a hacky solution to be fixed, since it doesn't seem to be a theme very used for Sensei currently.

<img width="1313" alt="Screenshot 2023-10-11 at 10 49 52" src="https://github.com/Automattic/sensei/assets/876340/220f1c4e-0b6b-4aba-aa23-dbf7a6dfbb4d">

Imran92 committed:
Add spacings for twentyone theme for login submit button

Imran92 commented:
> It seems it only fixes it for this specific notice, but the others using this method continue with the issue. I found https://github.com/Automattic/sensei/issues/6173. If it's not very easy to fix, maybe we could revert the commit and work on that in another PR?

I think we need to address the other notice issues in a separate PR as this PR is login-specific and there are good chances of side effects of any global changes.

> About Twenty Twenty-One, I still noticed that we don't have spacing on the buttons. But feel free to ignore this one if it's not easy or needs a hacky solution to be fixed, since it doesn't seem to be a theme very used for Sensei currently.

Right, I also don't think we probably shouldn't be too concerned about twenty twenty one theme issues. But I just added a little style here to add some space there only for twenty twenty one theme https://github.com/Automattic/sensei/pull/7204/commits/c8a3e278590095c622b28cf66c8e600992a0b0da

renatho reviewed (approved):
Looks good!! =)

> I think we need to address the other notice issues in a separate PR as this PR is login-specific and there are good chances of side effects of any global changes.

About this, I wonder if we should revert the commit with the fix we did specifically to this part. So when we fix it globally, it gets fixed the same way in all the places. WDYT?

Imran92 commented:
> About this, I wonder if we should revert the commit with the fix we did specifically to this part. So when we fix it globally, it gets fixed the same way in all the places. WDYT?

That's a very good thought @renatho  and I also thought about this. But the login form has its own function call to notice printing to make sure it is displayed immediately above the login form or login block https://github.com/Automattic/sensei/blob/7a5caf4cc30cf29f64ba8e72868fd32335bf4cb4/includes/class-sensei-frontend.php#L1033 which is different from other pages in general, separated from the global way of displaying notices. There's a chance that we'll still need to keep this change even then depending on how we decide to fix that. So for now, I'm keeping the current change to make it look not broken.
